### PR TITLE
Modernize JS and add more file loading from URLs

### PR DESCRIPTION
**Pull request by MariaNattestad**
This contains a lot of functionality. See commit messages for a full list -- they are very descriptive.

This PR is huge because I updated Ribbon to use modern bundling and ES6 imports -- basically all the imports are more robust, and libraries are imported explicitly instead of scripts sharing global variables.
Several libraries had to be updated to versions that are new enough to support this (e.g. D3 version 4 in 2016 was the first to support `import`, but this comes with other changes that we had to update a lot of code to deal with).

During testing for this, I also added URL file loading and the start to sessions, which enabled faster testing of the updated code. These just need a bit more extension to become full-fledged JSON session file loading functionality that we wanted.

And here are multi-bams loaded by URL:
Screenshot 2024-08-24 at 9 08 15 PM

### CODE REVIEW

File: js/index_splitthreader.js [hunk 1]
```diff
@@ -4243,12 +4376,233 @@ function open_variants_file() {
...
+    });
+    variants = variants.concat(records);
+  }
+
+  return variants;
+}
+
+async function open_variants_vcf_file() {
+  var raw_data;
+  var reader = new FileReader();
```
Comment (robertaboukhalil) [inline js/index_splitthreader.js:4402]: I think these two lines can be deleted?

File: js/index_splitthreader.js [hunk 2]
```diff
@@ -4190,6 +4315,22 @@ function genome_input_from_coverage(coverage_by_chromosome) {
...
+  mark_coverage_loaded_successfully();
+
+  run_everything_that_needs_coverage();
+  check_and_run_if_both_variants_and_coverage_loaded();
+
+}
+
 function open_coverage_file() {
   var raw_data;
   var reader = new FileReader();
```
Comment (robertaboukhalil) [inline js/index_splitthreader.js:4336]: This can be simplified as `const raw_data = await this.files[0].text()` to get the contents of the file

**Review by robertaboukhalil — APPROVED** [review]
Looks awesome!

RE: loading Aioli in both index_splitthreader.js and index_ribbon.js, we could load it once in a separate index.js that runs first and I think we should be able to import CLI from index.js

**Comment by MariaNattestad** [comment]
> Looks awesome!
> 
> RE: loading Aioli in both index_splitthreader.js and index_ribbon.js, we could load it once in a separate index.js that runs first and I think we should be able to import CLI from index.js

Yea I think we should pull up a few things for file loading into another script, including the new feature we will add soon to load a session JSON from a URL.

File: js/index_splitthreader.js (hunk 2 above)
Comment (MariaNattestad) [reply js/index_splitthreader.js]: oh nice, I have made that substitution now in about 10 places.

**Pull request merged**